### ISSUE TITLE
Printing XML CDATA section gives incorrect output

### ISSUE THREAD

**Issue by rdulmina** [ballerina-platform/ballerina-lang#32673]
**Description:**
Consider the bellow sample of XML CDATA added via #32564
```bal
public function main() {
    xml x7 = xml `<![CDATA[ abc --> <!-- --> some more text ]]>`;
    io:println(x7);
}
```
This prints `abc --&amp;gt; &amp;lt;!-- --&amp;gt; some more text` where it should print `abc --&gt; &lt;!-- --&gt; `some more text

**Affected Versions:**
Current master

**Comment by HindujaB** [comment]
This sample code  provides the correct output for `toString()` call in current master. 
```ballerina
public function main() {
    xml x7 = xml `<![CDATA[ abc --> <!-- --> some more text ]]>`;
    io:println(x7.toString() == " abc --&gt; &lt;!-- --&gt; some more text ");  // gives true
}
```
@rdulmina Shall we check this with current master?

**Comment by rdulmina** [comment]
> This sample code provides the correct output for `toString()` call in current master.
> 
> ```ballerina
> public function main() {
>     xml x7 = xml `<![CDATA[ abc --> <!-- --> some more text ]]>`;
>     io:println(x7.toString() == " abc --&gt; &lt;!-- --&gt; some more text ");  // gives true
> }
> ```
> 
> @rdulmina Shall we check this with current master?

Yeah seems to be fixed. Let's add a test case and close this :-)

**Comment by HindujaB** [comment]
The error is happening when we have this inside a unit test. This is due to the stream writer usage in the `stringValue()` implementation. 
Fixed with PR #32712
